Plugins/BTCPayServer.Plugins.ShopifyPlugin/Controllers/UIShopifyV2Controller.cs

```diff
@@ -236,72 +236,81 @@ public async Task<IActionResult> Settings(string storeId,
+        var store = await _storeRepo.FindStore(storeId);
+        if (order is null || store is null)
+            return BadRequest("Invalid checkout token");
 
-		var orderId = order.Id.Id;
-		var searchTerm = $"{Extensions.SHOPIFY_ORDER_ID_PREFIX}{orderId}";
-		var invoices = await _invoiceRepository.GetInvoices(new InvoiceQuery()
-		{
-			TextSearch = searchTerm,
-			StoreId = new[] { storeId }
-		});
+        var containsKeyword = order.PaymentGatewayNames.Any(pgName => ShopifyHostedService.IsBTCPayServerGateway(pgName));
+        if (!containsKeyword)
+            return BadRequest("Order wasn't fulfiled with BTCPay Server payment option");
 
-		// This prevent a race condition where two invoices get created for same order
-		using var l = await OrderLocks.LockAsync(orderId, cancellationToken);
+        var orderId = order.Id.Id;
+        var searchTerm = $"{Extensions.SHOPIFY_ORDER_ID_PREFIX}{orderId}";
+        var invoices = await _invoiceRepository.GetInvoices(new InvoiceQuery()
+        {
+            TextSearch = searchTerm,
+            StoreId = new[] { storeId }
+        });
 
-		var orderInvoices = 
-			invoices.Where(e => e.GetShopifyOrderId() == orderId).ToArray();
-		var currentInvoice =  orderInvoices.FirstOrDefault();
-		if (currentInvoice != null)
-			return RedirectToInvoiceCheckout(currentInvoice.Id);
+        // This prevent a race condition where two invoices get created for same order
+        using var l = await OrderLocks.LockAsync(orderId, cancellationToken);
 
-		var baseTx = order.Transactions.FirstOrDefault(t => t is { Kind: "SALE", ManuallyCapturable: true });
-		if (baseTx is null)
-			return BadRequest("The shopify order is not capturable");
-		var settings = await _storeRepo.GetSettingAsync<ShopifyStoreSettings>(storeId, ShopifyStoreSettings.SettingsName);
-		var amount = order.TotalOutstandingSet.PresentmentMoney;
-		var invoice = await _invoiceController.CreateInvoiceCoreRaw(
-				new CreateInvoiceRequest()
-				{
-					Amount = amount.Amount,
-					Currency = amount.CurrencyCode,
-					Metadata = new JObject
-					{
-						["orderId"] = order.Name,
-						["orderUrl"] = GetOrderUrl(settings?.Setup?.ShopUrl, orderId),
-						["shopifyOrderId"] = orderId,
-						["shopifyOrderName"] = order.Name,
-						["gateway"] = baseTx.Gateway
-					},
-					AdditionalSearchTerms =
-					[
-						order.Name,
-						orderId.ToString(CultureInfo.InvariantCulture),
-						searchTerm
-					],
-					Checkout = new()
-					{
-						RedirectURL = order.StatusPageUrl
-					}
-				}, store,
-				Request.GetAbsoluteRoot(), [searchTerm], cancellationToken);
-		return RedirectToInvoiceCheckout(invoice.Id);
-	}
+        var orderInvoices =
+            invoices.Where(e => e.GetShopifyOrderId() == orderId).ToArray();
+        var currentInvoice = orderInvoices.FirstOrDefault();
+        if (currentInvoice != null)
+            return redirect ? RedirectToInvoiceCheckout(currentInvoice.Id) : Ok();
```

Comment:
Could have used the `Content-Type` instead of an additional parameter, but that's fine, can let it like that.